packages/contracts-bedrock/test/LivenessModule.t.sol

```diff
@@ -0,0 +1,512 @@
+
+        _warpPastLivenessInterval();
+        _removeAnOwner(ownerToRemove, safeInstance.owners);
+
+        assertFalse(safeInstance.safe.isOwner(ownerToRemove));
+        assertEq(safeInstance.safe.getOwners().length, ownersBefore - 1);
+    }
+
+    /// @dev Tests if removing all owners works correctly
+    function test_removeOwners_allOwners_succeeds() external {
+        uint256 numOwners = safeInstance.owners.length;
+
+        address[] memory ownersToRemove = new address[](numOwners);
+        for (uint256 i = 0; i < numOwners; i++) {
+            ownersToRemove[i] = safeInstance.owners[i];
+        }
+        address[] memory prevOwners = safeInstance.getPrevOwners(ownersToRemove);
+
+        _warpPastLivenessInterval();
+        livenessModule.removeOwners(prevOwners, ownersToRemove);
+        assertEq(safeInstance.safe.getOwners().length, 1);
+        assertEq(safeInstance.safe.getOwners()[0], fallbackOwner);
+        assertEq(safeInstance.safe.getThreshold(), 1);
+    }
+}
+
+/// @dev A copy of LivenessModule.get75PercentThreshold as a free function to use below.
+function get75PercentThreshold(uint256 _numOwners) pure returns (uint256 threshold_) {
+    threshold_ = (_numOwners * 75 + 99) / 100;
+}
+
+contract LivenessModule_RemoveOwnersFuzz_Test is LivenessModule_TestInit {
+    using SafeTestLib for SafeInstance;
+
+    /// @dev We put this array in storage so that we can more easily populate it using push in the tests below.
+    address[] ownersToRemove;
+
+    /// @dev Options for handling the event that the number of owners remaining is less than minOwners
+    enum ShutDownBehavior {
+        Correct, // Correctly removes the owners and transfers to the shutDown owner
+        DoesNotTransferToFallbackOwner, // Removes all but one owner, and does not transfer to the shutDown owner
+        DoesNotRemoveAllOwners // Leaves more than one owner when below minOwners
+    }
+
+    /// @dev This contract inherits the storage layout from the LivenessModule_TestInit contract, but we
+    ///      override the base setUp function, to avoid instantiating an unnecessary Safe and liveness checking system.
+    function setUp() public override {
+        vm.warp(initTime);
+        fallbackOwner = makeAddr("fallbackOwner");
+    }
+
+    /// @dev Extracts the setup of the test environment into a separate function.
+    function _prepare(
+        uint256 _numOwners,
+        uint256 _minOwners,
+        uint256 _numLiveOwners
+    )
+        internal
+        returns (uint256 numOwners_, uint256 minOwners_, uint256 numLiveOwners_)
+    {
+        // First we modify the test parameters to ensure that they describe a plausible starting point.
+        //
+        // _numOwners must be at least 4, so that _minOwners can be set to at least 3 by the following bound() call.
+        // Limiting the owner set to 20 helps to keep the runtime of the test reasonable.
+        console.log("bounding numOwners");
```

Comment:
dead consoles?

Reply:
I think these are going to be useful for anyone dealing with this test in the future, and they don't show up if you don't add `-v`. We don't mind using `bound` which comes with a console